src/arrayMethodJoin.js

```diff
@@ -4,8 +4,32 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let stringAfterJoin = '';
+    const copyArray = this.slice();
```

Comment:
2. [CODE STYLE] - avoid creating needless copies of arrays, you can iterate existing array